improve `$header` modifier description (#234)
AdguardTeam/KnowledgeBase: docs/general/ad-filtering/create-own-filters.md

* `@@||example.com^$header=set-cookie:/foo\, bar\$/` unblocks requests whose responses have the `Set-Cookie` header with value matching the `foo, bar$` regular expression.
* `@@||example.com^$header=set-cookie` unblocks requests whose responses have a `Set-Cookie` header with any value.

:::caution Limitations

> In Adguard Browser Extension, [`$header`](#header-modifier) modifier is only compatible with [`$csp`](#csp-modifier) and [`$removeheader`](#removeheader-modifier) advanced modifiers. Rules that use [`$header`](#header-modifier) modifier with one of these, are subject to allowlisting logic of advanced modifiers.

Reviewer: We should explain the difference in matching logic of this modifier. It is matched only when headers are received so if the request was blocked or redirected on an earlier stage nothing will be applied. This logic is similar for both the browser extension and CoreLibs.

:::

:::info Compatibility

> Rules with the `$header` modifier are supported by AdGuard for Windows, Mac, and Android, **running CoreLibs version 1.11 or later**, and AdGuard Browser Extension, **running TSUrlFilter vX.X.X or later**.

Reviewer: So?

Author: This is waiting for an actual version num on tsurlfilter 2.2 release

- `@@||example.com^$header=set-cookie:/foo\, bar\$/` unblocks requests whose responses have the `Set-Cookie` header with value matching the `foo, bar$` regular expression.
- `@@||example.com^$header=set-cookie` unblocks requests whose responses have a `Set-Cookie` header with any value.

##### `$header` modifier limitations {#header-modifier-limitations}

:::caution Restrictions

1. The `$header` modifier can be matched only when headers are received.

>   So if the request is blocked or redirected on an earlier stage, the modifier cannot be applied.

Reviewer 2: ```suggestion
  So if the request is blocked or redirected at an earlier stage, the modifier cannot be applied.
```

- `||domain.com^$popup` — if you try to go to `http://domain.com/` from any page in the browser, a new tab in which specified site has to be opened will be closed by this rule.

##### `$popup` modifier limitations {#popup-modifier-limitations}

:::caution Limitations

> 1. `$popup` modifier works best in AdGuard Browser Extension.

Reviewer 2: ```suggestion
1. The `$popup` modifier works best in AdGuard Browser Extension.
```

1. In AdGuard for iOS and AdGuard for Safari, `$popup` rules simply block the page right away.

> 1. In AdGuard for Windows, AdGuard for Mac, and AdGuard for Android, `$popup` modifier may not detect a popup

Reviewer 2: ```suggestion
1. In AdGuard for Windows, AdGuard for Mac, and AdGuard for Android, the `$popup` modifier may not detect a popup
```

  in some cases and it will not be blocked.

>   `$popup` modifier applies the `document` content type with a special flag which is passed to a blocking page.

Reviewer 2: ```suggestion
  Thу `$popup` modifier applies the `document` content type with a special flag that is passed to a blocking page.
```

  Blocking page itself can do some checks and close the window if it is really a popup.
  Otherwise, page should be loaded.
  It can be combined with other request type modifiers, such as `$third-party` and `$important`.

:::

:::info Compatibility

> Rules with `$popup` modifier are not supported by AdGuard Content Blocker.

Reviewer 2: ```suggestion
Rules with the `$popup` modifier are not supported by AdGuard Content Blocker.
```
